src/aa_remove_data/pb_utils.py

```diff
@@ -0,0 +1,164 @@
+from datetime import datetime, timedelta
+from os import PathLike
+
+from aa_remove_data.generated import EPICSEvent_pb2
+
+
+class PBUtils:
+    def __init__(self, filepath: PathLike | None = None):
+        """Initialise a PBUtils object. If filepath is set, read the protobuf
+        file at this location to gether its header, samples and type.
+
+        Args:
+            filepath (Optional[PathLike], optional): Path to pb file to be
+            read. Defaults to None.
+        """
+        self.header = EPICSEvent_pb2.PayloadInfo()  # type: ignore
+        self.samples = []
+        self.sample_type = ""
+        if filepath:
+            self.read_pb(filepath)
+
+    def _escape_data(self, data: bytes) -> bytes:
+        """Replace escape characters with alternatives, to avoid conflitcs.
+        Should exactly mirror _unescape_data().
+
+        Args:
+            data (bytes): A serialised protobuf sample.
+
+        Returns:
+            bytes: The serialised sample with escape characters replaced.
+        """
+        data = data.replace(b"\x1b", b"\x1b\x01")  # Escape escape character
+        data = data.replace(b"\x0a", b"\x1b\x02")  # Escape newline
+        data = data.replace(b"\x0d", b"\x1b\x03")  # Escape carriage return
+        return data
+
+    def _unescape_data(self, data: bytes) -> bytes:
```

Comment:
I understand the logic of this function name, but it doesn't make much sense in practice. Consider something more specific to the application, such as `remove/add_aa_escape_chars` or `add/remove_sample_per_line`. (feel free to add something else if it fits better)